Core/include/Acts/Utilities/TrackHelpers.hpp

```diff
@@ -401,6 +402,97 @@ void calculateTrackQuantities(track_proxy_t track) {
   }
 }
 
+/// Helper function to trim track states from the front of a track
+/// @tparam track_proxy_t the track proxy type
+/// @param track the track to trim
+/// @param trimHoles whether to trim holes
+/// @param trimOutliers whether to trim outliers
+/// @param trimMaterial whether to trim pure material states
+template <TrackProxyConcept track_proxy_t>
+void trimTrackFront(track_proxy_t track, bool trimHoles, bool trimOutliers,
+                    bool trimMaterial)
+  requires(!track_proxy_t::ReadOnly)
+{
+  using TrackStateProxy = typename track_proxy_t::TrackStateProxy;
+
+  // TODO specialize if track is forward linked
+
+  std::optional<TrackStateProxy> front;
+
+  for (TrackStateProxy trackState : track.trackStatesReversed()) {
+    TrackStateType typeFlags = trackState.typeFlags();
+    if (trimHoles && typeFlags.test(TrackStateFlag::HoleFlag)) {
+      continue;
+    }
+    if (trimOutliers && typeFlags.test(TrackStateFlag::OutlierFlag)) {
+      continue;
+    }
+    if (trimMaterial && typeFlags.test(TrackStateFlag::MaterialFlag) &&
```

Comment:
This will trim holes, outliers, and pure material. The [old behaviour](https://github.com/acts-project/acts/pull/3391/files#diff-55dd901875f5a87b6a9fee495de245296efbfd45aa9f9483036bcd299a277e09L855) skipped everything except *material* (not measurements, as I would have expected). This is clearly different. I think the new behaviour makes more sense, but this might explain the differences we see.

Reply:
Good point!

Reply:
- https://github.com/acts-project/acts/pull/3648